src/audio/copier/copier.c

```diff
@@ -990,6 +977,29 @@ static int copier_get_hw_params(struct comp_dev *dev, struct sof_ipc_stream_para
 	return dai_common_get_hw_params(dd, dev, params, dir);
 }
 
+static int copier_bind(struct processing_module *mod, void *data)
+{
+	struct copier_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
+	struct ipc4_module_bind_unbind *bu = data;
+	struct list_item *sink_list;
+	struct comp_buffer *sink;
+
+	/* update corresponding sink format */
+	list_for_item(sink_list, &dev->bsink_list) {
+		int sink_id;
+
+		sink = container_of(sink_list, struct comp_buffer, source_list);
+		sink_id = IPC4_SRC_QUEUE_ID(buf_get_id(sink));
+		if (sink_id == bu->extension.r.src_queue) {
+			ipc4_update_buffer_format(sink, &cd->out_fmt[sink_id]);
+			break;
```

Comment:
you could use `return 0;` here and return a `-ENODEV` error below, which would mean, that no suitable sink ID was found

Reply:
If we return an error lets make sure we an error entry in the log to help debug. Thanks !

Reply:
Sure. Thanks!